consensus/safety-rules/src/tests/suite.rs

```diff
@@ -600,3 +607,144 @@ fn test_sign_proposal_with_early_preferred_round(func: Callback) {
         Error::InvalidQuorumCertificate("Preferred round too early".into())
     );
 }
+
+fn test_uninitialized_signer(func: Callback) {
+    // Testing for an uninitialized Option<ValidatorSigner>
+
+    let (mut safety_rules, signer) = func();
+
+    let (proof, genesis_qc) = make_genesis(&signer);
+    let round = genesis_qc.certified_block().round();
+
+    let a1 = test_utils::make_proposal_with_qc(round + 1, genesis_qc, &signer);
+    let err = safety_rules.update(a1.block().quorum_cert()).unwrap_err();
+    assert_eq!(err, Error::NotInitialized);
+
+    safety_rules.initialize(&proof).unwrap();
+    assert!(safety_rules.update(a1.block().quorum_cert()).is_ok());
+}
+
+#[allow(dead_code)]
```

Comment:
Please respond to this comment rather than resolving it without response. In general, for discussions, you should let the one that opened the discussion to resolve it.

On the other hand, if it is a feedback such as rename this to that or use a filter_map instead of a map + filter then feel free to resolve yourself.

Reply:
Plan added in comment of the code and is_ok() refactored.

Reply:
panic is not the right word

@zekun000, if our key isn't in the list, do we want to keep running or panic? The panic happens in the consensus code, which expects safety rules to update successfully.

@zaxguo, even if we do not want to panic, we can still return an enum on the Ok that implies whether or not we are present in the list.

Reply:
perhaps we can have a flag in consensus state that we can check to see if this node is in the validator set that way you can write a proper test here.

Reply:
> perhaps we can have a flag in consensus state that we can check to see if this node is in the validator set that way you can write a proper test here.

This is execellent advice. I adopted it in the latest commit, which greatly simply one test_validator_not_in_set.